[nrf noup] Added workaround for missing Debug symbols in tfm_s image (#62)
nrfconnect/sdk-trusted-firmware-m: secure_fw/CMakeLists.txt

        ${COMPILER_CP_FLAG}
)

# To get debug symbols of function called through veneer functions
file(STRINGS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_symbols.txt VENEER_SYMBOLS)

> 

Reviewer: Sorry, bad CMake on my end.

we need list(APPEND veneer_link_options -Wl,--undefined=${function})

> 

Author: Oh yeah makes more sense

        ${COMPILER_CP_FLAG}
)

> if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")

Reviewer: We want to apply this workaround for all build types.

I want to be able to debug non-Debug builds as well.

e.g. Debug does all kinds of things and may increase the binary size such that I can't use Debug to debug my app. In which case I wish to debug the non-Debug build. And I want debug symbols for this.

>     # To get debug symbols of function called through veneer functions

Reviewer: Let us give a detailed description of what the problem is and how we are solving it here.

It should include at least everything relevant from the commit message.

It should also include a description of why this does not cause bloat.

And be honest about that we are uncertain of the root cause of the issue and that we suspect a bug in LD.

>     file(STRINGS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_symbols.txt VENEER_SYMBOLS)

Reviewer: ```suggestion
    file(STRINGS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_symbols.txt veneer_symbols)
```

lower-case is (generally, not sure about TF-M) used for locally scoped variables like VENEER_SYMBOOLS.

        ${COMPILER_CP_FLAG}
)

# This is a workaround to add debug symbols for the veneer functions of the
# secure part of TF-M. The problem is that the linker strips away the debug
# section of the functions called by a veneer function. The root cause is not
# fully clear, but might be a bug in the linker as it removes the debug
# sections with the reason of being unused but keeps the actual .text section
# of the functions. To work around this we instruct the linker to expilicitly
# keep the, "unused" functions.
if(EXISTS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_functions.txt)
    file(STRINGS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_functions.txt veneer_functions)
else()
    set(veneer_functions "")
endif()

foreach(function ${veneer_functions})
    list(APPEND veneer_link_options -Wl,--undefined=${function})
endforeach()

target_link_options(tfm_s
    PRIVATE
        --entry=Reset_Handler

>         $<$<C_COMPILER_ID:GNU>:-Wl,-Map=${CMAKE_BINARY_DIR}/bin/tfm_s.map> 

Reviewer 2: FYI: There is a trailing whitespace her, might be caught be the checkpatch / nit scripts?

        ${COMPILER_CP_FLAG}
)

# This is a workaround to add debug symbols for the veneer functions of the
# secure part of TF-M. The problem is that the linker strips away the debug
# section of the functions called by a veneer function. The root cause is not
# fully clear, but might be a bug in the linker as it removes the debug sections
# with the reason of beeing unused but keeps the actual .text section of the functions

> # This tells the linker to expilicitly keep the, "unused" functions.

Reviewer: ```suggestion
# To work around this we instruct the linker to expilicitly keep the, "unused" functions.
```

The word "This" sounds like we are refering to something we just described.

        ${COMPILER_CP_FLAG}
)

# This is a workaround to add debug symbols for the veneer functions of the
# secure part of TF-M. The problem is that the linker strips away the debug
# section of the functions called by a veneer function. The root cause is not
# fully clear, but might be a bug in the linker as it removes the debug 

> # sections with the reason of beeing unused but keeps the actual .text section 

Reviewer 3: ```suggestion
# sections with the reason of being unused but keeps the actual .text section 
``` 
Lets leave the bees alone

Author: But I want more honey...

        ${COMPILER_CP_FLAG}
)

# This is a workaround to add debug symbols for the veneer functions of the
# secure part of TF-M. The problem is that the linker strips away the debug
# section of the functions called by a veneer function. The root cause is not
# fully clear, but might be a bug in the linker as it removes the debug
# sections with the reason of being unused but keeps the actual .text section
# of the functions. To work around this we instruct the linker to expilicitly
# keep the, "unused" functions.
if(EXISTS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_functions.txt)
    file(STRINGS ${CMAKE_BINARY_DIR}/generated/tools/tfm_veneers_functions.txt veneer_functions)
else()
    set(veneer_functions "")
endif()

foreach(function ${veneer_functions})
    list(APPEND veneer_link_options -Wl,--undefined=${function})
endforeach()

target_link_options(tfm_s
    PRIVATE

>     --entry=Reset_Handler

Reviewer 2: Is it intentional to change the indentation here?
It looks different from the rest of the source file.

Author: No just and stupid autoformat from my vs code :(
